kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/KieContainerResourceFilter.java

```diff
@@ -0,0 +1,189 @@
+    private final ReleaseIdFilter releaseIdFilter;
+
+    @XmlElement(name = "container-status-filter")
+    @XStreamAlias("container-status-filter")
+    private final KieContainerStatusFilter statusFilter;
+
+    private KieContainerResourceFilter() {
+        // needed for JAXB
+        this.releaseIdFilter = ReleaseIdFilter.ACCEPT_ALL;
+        this.statusFilter = KieContainerStatusFilter.ACCEPT_ALL;
+    }
+
+    public KieContainerResourceFilter(ReleaseIdFilter releaseIdFilter, KieContainerStatusFilter statusFilter) {
+        this.releaseIdFilter = releaseIdFilter;
+        this.statusFilter = statusFilter;
+    }
+
+    public KieContainerResourceFilter(ReleaseIdFilter releaseIdFilter) {
+        this(releaseIdFilter, KieContainerStatusFilter.ACCEPT_ALL);
+    }
+
+    public ReleaseIdFilter getReleaseIdFilter() {
+        return releaseIdFilter;
+    }
+
+    public KieContainerStatusFilter getStatusFilter() {
+        return statusFilter;
+    }
+
+    public boolean accept(KieContainerResource kieContainerResource) {
+        if (kieContainerResource == null) {
+            throw new IllegalArgumentException("KieContainerResource can not be null!");
+        }
+        // in case resolved release id exists, check against that
+        ReleaseId resolvedReleaseId = kieContainerResource.getResolvedReleaseId();
+        if (resolvedReleaseId != null) {
+            if (!releaseIdFilter.accept(resolvedReleaseId)) {
+                return false;
+            }
+        } else {
+            if (!releaseIdFilter.accept(kieContainerResource.getReleaseId())) {
+                return false;
+            }
+        }
+        KieContainerStatus status = kieContainerResource.getStatus();
+        if (status != null && !statusFilter.accept(status)) {
+            return false;
+        }
+        // all sub-filters accepted the container, so it is a match
+        return true;
+    }
+
+    /**
+     * Creates representation of this filter which can be used as part of the URL (e.g. the "?" query part).
+     *
+     * @return string representation that can be directly used in URL (as query params), without the leading '?'
+     */
+    public String toURLQueryString() {
+        StringBuilder urlQuerySB = new StringBuilder();
+        if (releaseIdFilter.getGroupId() != null) {
+            urlQuerySB.append("groupId=").append(releaseIdFilter.getGroupId());
+        }
+        if (releaseIdFilter.getArtifactId() != null) {
+            urlQuerySB.append("&artifactId=").append(releaseIdFilter.getArtifactId());
+        }
+        if (releaseIdFilter.getVersion() != null) {
+            urlQuerySB.append("&version=").append(releaseIdFilter.getVersion());
+        }
+        // don't send over the default filter (e.g. one that accepts all the states)
+        if (!statusFilter.equals(KieContainerStatusFilter.ACCEPT_ALL)) {
+            urlQuerySB.append("&status=").append(join(statusFilter.getAcceptedStatuses(), ","));
```

Comment:
Thanks for the tip. I did not use Java 8 API because we will likely need to cherry-pick this into 6.5.x. But thinking about this again, the StringJoiner is much nicer, so I will use it on master.